### PR TITLE
feature/448_deprecate_xml_notifications

### PR DESCRIPTION
**Pull request by pcoello25**
* Fix issue #448 
* 100% unit test passed: 
```
Results : Tests run: 88, Failures: 0, Errors: 0, Skipped: 0
```
* Assignee: @frbattid

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -3,3 +3,4 @@
 - [HARDENING] Use precompiled regexes in Utils.encode() and Utils.encodeHive() (#818)
 - [HARDENING] Remove a temporal fix regarding Orion <= 0.10.0 in OrionRESTHandler (#840) 
 - [HARDENING] Add an efficient function for converting ArrayList to String in Utils.java (#841)
+- [FEATURE] Remove XML notifications in all sinks and in the documentation (#448)
```
Comment (frbattid) [inline CHANGES_NEXT_RELEASE]: Remove support for XML notifications...

Comment (pcoello25) [inline CHANGES_NEXT_RELEASE]: Fixed in 5ded8ef

**Comment by pcoello25** [comment]
Adding some examples:
* Sending a notification in json format:
```
time=2016-03-07T09:50:09.144CET | lvl=DEBUG | trans= | srv= | subsrv= | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[187] : Header content-type received with value application/json
time=2016-03-07T09:50:09.144CET | lvl=DEBUG | trans= | srv= | subsrv= | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[187] : Header accept received with value application/json
time=2016-03-07T09:50:09.148CET | lvl=INFO | trans=1457340596-803-0000000000 | srv=service | subsrv=servicePath | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[247] : Received data ({    "subscriptionId": "51c0ac9ed714fb3b37d7d5a8",    "originator": "localhost",    "contextResponses": [         {             "contextElement": {                 "attributes": [                     {                         "name": "pressure",                         "type": "float",                         "value": "12.3"                    }                 ],                 "type": "Room",                 "isPattern": "false",                 "id": "Room.619"            },             "statusCode": {                 "code": "200",                 "reasonPhrase": "OK"            }         },         {             "contextElement": {                 "attributes": [                     {                         "name": "humidity",                         "type": "float",                         "value": "3.21"                    }                ],                 "type": "Room",                 "isPattern": "false",                 "id": "Room.22"            },             "statusCode": {                 "code": "200",                 "reasonPhrase": "OK"            }         }     ] })
time=2016-03-07T09:50:12.126CET | lvl=INFO | trans=1457340596-803-0000000000 | srv=service | subsrv=servicePath | function=persistAggregation | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionMySQLSink[470] : [mysql-sink] Persisting data at OrionMySQLSink. Database (service), Table (rooms_allrooms), Fields ((recvTimeTs,recvTime,fiwareServicePath,entityId,entityType,attrName,attrType,attrValue,attrMd)), Values (('1457340609150','2016-03-07T08:50:09.150Z','rooms','Room.619','Room','pressure','float','12.3','[]'),('1457340609150','2016-03-07T08:50:09.150Z','rooms','Room.22','Room','humidity','float','3.21','[]'))
```

* Sending a notification in xml format:
```
time=2016-03-07T09:57:06.059CET | lvl=DEBUG | trans=1457340596-803-0000000000 | srv=service | subsrv=servicePath | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[187] : Header content-type received with value application/xml
time=2016-03-07T09:57:06.059CET | lvl=WARN | trans=1457340596-803-0000000000 | srv=service | subsrv=servicePath | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[191] : Bad HTTP notification (application/xml content type not supported)
time=2016-03-07T09:57:06.061CET | lvl=WARN | trans=1457340596-803-0000000000 | srv=service | subsrv=servicePath | function=doPost | comp=Cygnus | msg=org.apache.flume.source.http.HTTPSource$FlumeHTTPServlet[186] : Received bad request from client. 
org.apache.flume.source.http.HTTPBadRequestException: application/xml content type not supported
	at com.telefonica.iot.cygnus.handlers.OrionRestHandler.getEvents(OrionRestHandler.java:192)
	at org.apache.flume.source.http.HTTPSource$FlumeHTTPServlet.doPost(HTTPSource.java:184)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:725)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:814)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:401)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:766)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.content(HttpConnection.java:945)
	at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:756)
	at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:218)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at org.mortbay.jetty.bio.SocketConnector$Connection.run(SocketConnector.java:228)
	at org.mortbay.thread.QueuedThreadPool$PoolThread.run(QueuedThreadPool.java:582)
```
The logs above belongs to the localhost response:
```
* upload completely sent off: 714 out of 714 bytes
< HTTP/1.1 400 Bad request from client. application/xml content type not supported
< Content-Type: text/html; charset=iso-8859-1
< Cache-Control: must-revalidate,no-cache,no-store
< Content-Length: 1487
< Server: Jetty(6.1.26)
< 
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"/>
<title>Error 400 Bad request from client. application/xml content type not supported</title>
</head>
<body><h2>HTTP ERROR 400</h2>
<p>Problem accessing /notify. Reason:
<pre>    Bad request from client. application/xml content type not supported</pre></p><hr /><i><small>Powered by Jetty://</small></i><br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                
<br/>                                                

</body>
</html>
* Connection #0 to host localhost left intact
```
* Sending a grouping-rule in json format:
```
time=2016-03-07T09:49:57.100CET | lvl=INFO | trans= | srv= | subsrv= | function=<init> | comp=Cygnus | msg=com.telefonica.iot.cygnus.interceptors.GroupingRules[62] : Grouping rules read: {    "grouping_rules": [        {            "id": 1,            "fields": [                "entityType"            ],            "regex": "Room",            "destination": "allrooms",            "fiware_service_path": "rooms"            "sinks": [		"mysql-sink"            ]        }   ]}
time=2016-03-07T09:49:57.110CET | lvl=INFO | trans= | srv= | subsrv= | function=<init> | comp=Cygnus | msg=com.telefonica.iot.cygnus.interceptors.GroupingRules[72] : Grouping rules syntax is OK
time=2016-03-07T09:49:57.111CET | lvl=INFO | trans= | srv= | subsrv= | function=<init> | comp=Cygnus | msg=com.telefonica.iot.cygnus.interceptors.GroupingRules[76] : Grouping rules regex'es have been compiled
time=2016-03-07T09:49:57.112CET | lvl=INFO | trans= | srv= | subsrv= | function=<init> | comp=Cygnus | msg=com.telefonica.iot.cygnus.interceptors.GroupingRules[62] : Grouping rules read: {    "grouping_rules": [        {            "id": 1,            "fields": [                "entityType"            ],            "regex": "Room",            "destination": "allrooms",            "fiware_service_path": "rooms"            "sinks": [		"mysql-sink"            ]        }   ]}
time=2016-03-07T09:49:57.112CET | lvl=INFO | trans= | srv= | subsrv= | function=<init> | comp=Cygnus | msg=com.telefonica.iot.cygnus.interceptors.GroupingRules[72] : Grouping rules syntax is OK
time=2016-03-07T09:49:57.113CET | lvl=INFO | trans= | srv= | subsrv= | function=<init> | comp=Cygnus | msg=com.telefonica.iot.cygnus.interceptors.GroupingRules[76] : Grouping rules regex'es have been compiled
```

Let me see the behaviour of that duplicated log..

**Comment by frbattid** [comment]
LGTM

**Pull request merged**